### PR TITLE
Added code to hide internal indexes when searching logs

### LINKED ISSUE #89: Hide internal indexes when searching logs
# Description
Update `ExpandAndReturnIndexNames()` in resolveIndexHandler.go so that the internal indexes are hidden when searching logs. These indexes are:
* traces
* red-traces
* service-dependency

# Goals
* Update `ExpandAndReturnIndexNames()` as mentioned.
* Update the unit tests in resolveIndexHandler_test.go to test your changes.
* Verify that when you ingest some traces (see the description of PR #38 for instructions) and logs and then go to the siglens UI -> Logs, in the Index dropdown you see the indexes for the logs but don't see any of the internal trace indexes. If you see the internal indexes, you should make any additional changes so that those are hidden

### PR DESCRIPTION
**Pull request by ashutoxh**
# Description
Added code to hide internal indexes when searching logs. 
These indexes are: 
  - traces
  - red-traces
  - service-dependency

Added test case for ExpandAndReturnIndexNames in virtualtable_test.go

Fixes #89

# Testing
Check the UI dropdown of Index, none of the internal indexes were present.

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [x] I have self-reviewed this PR.
- [x] I have removed all print-debugging and commented-out code that should not be merged.
- [x] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [x] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: pkg/es/reader/resolveIndexHandler.go [hunk 1]
```diff
@@ -70,6 +75,9 @@ func ExpandAndReturnIndexNames(indexPattern string, allVirtualTableNames map[str
 		}
 
 	} else {
+		if isIndexExcluded(indexPattern) {
```
Comment (sunitakawane) [inline pkg/es/reader/resolveIndexHandler.go:78]: remove this part. We want to exclude internal indexes only when we pass "*" as index value. If we send specific index name (traces/red-traces/service-dependency) we should search against the exact index names.

File: pkg/es/reader/resolveIndexHandler_test.go [hunk 2]
```diff
@@ -146,6 +155,33 @@ func Test_ExpandAndReturnIndexNames(t *testing.T) {
 	assert.EqualValues(t, indicesExpected, indicesEntries, fmt.Sprintf("Comparison failed, expected=%v, actual=%v", indicesExpected, indicesEntries))
 	assert.EqualValues(t, aliasesExpected, aliasesEntries, fmt.Sprintf("Comparison failed, expected=%v, actual=%v", aliasesExpected, aliasesEntries))
 
+	indexPattern = "service-dependency*"
```
Comment (sunitakawane) [inline pkg/es/reader/resolveIndexHandler_test.go:158]: make it indexPattern = "service-dependency"

**Review by sunitakawane — CHANGES_REQUESTED** [review]
With current changes , periodic job for traces (ProcessRedTracesIngest,MakeTracesDependancyGraph) are causing panic since we are always ignoring tracing related indexes.

File: pkg/es/reader/resolveIndexHandler.go (hunk 1 above)
Comment (ashutoxh) [reply pkg/es/reader/resolveIndexHandler.go:78]: This is true just for the API part that calls resolveIndexHandler.go or for the UI part [here](https://github.com/ashutoxh/siglens/blob/develop/pkg/virtualtable/virtualtable.go#L606) as well?

File: pkg/virtualtable/virtualtable.go [hunk 3]
```diff
@@ -589,14 +591,21 @@ func ExpandAndReturnIndexNames(indexNameIn string, orgid uint64, isElastic bool)
...
 			finalResultsMap[indexName] = true
 		}
 	} else {
 		indexNames := strings.Split(indexNameIn, ",")
 		for _, indexName := range indexNames {
+			if isIndexExcluded(indexName) {
+				continue
+			}
 			if strings.Contains(indexName, "*") {
 				regexStr := "^" + strings.ReplaceAll(indexName, "*", `.*`) + "$"
```
Comment (sunitakawane) [inline pkg/virtualtable/virtualtable.go:610]: move the above if loop `if isIndexExcluded(indexName) {..} `inside the loop `if strings.Contains(indexName, "*")`

File: pkg/es/reader/resolveIndexHandler.go (hunk 1 above)
Comment (sunitakawane) [reply pkg/es/reader/resolveIndexHandler.go:78]: Please see this https://github.com/siglens/siglens/pull/119/files#r1403638696

**Review by sunitakawane — APPROVED** [review]

**Pull request merged**